src/transform/affine.rs

```diff
@@ -1,24 +1,95 @@
+use super::Transform;
 use ndarray::{array, prelude::*};
+use ndarray_linalg::Inverse;
 
+
+pub fn transform_from_2dmatrix(in_array: Array2<f64>) -> AffineTransform {
+    /// converts a matrix into an equivalent Transform
+    let transform = match in_array.inv() {
+        Ok(inv) => AffineTransform {
+            matrix2d_transform: in_array.clone(),
+            matrix2d_transform_inverse: inv,
+            inverse_exists: true,
+        },
+        Err(e) => AffineTransform {
+            matrix2d_transform: in_array.clone(),
+            matrix2d_transform_inverse: Array2::zeros((2, 2)),
+            inverse_exists: false,
+        },
+    };
+    return transform;
+}
+
+pub struct AffineTransform {
+    /// Represents a linear transform of either size 2x2
```

Comment:
Ditto, I imagine this comment should be above the struct? The idea is these should show up for users when they look at the docs (so run `cargo doc` and have a peruse to see how it looks)